pages/today.vue

```diff
@@ -1,11 +1,39 @@
 <script setup lang="ts">
-import { format } from 'date-fns';
+const user = useSupabaseUser();
 
 const date = Date.now();
+const userId = user.value!.id;
```

Comment:
I have a question. Is it guaranteed that `user.value` always exist here? Users cannot enter this page before authentication? I wish to have something so that you don't have to put `!` here. (it can be done in a separate PR, tho. totally not blocking the merge of this PR)

Reply:
@eunjae-lee Yes - `user.value` is guaranteed here because I have the `nuxt.config.ts` set up as the following for supabase so that authentication always takes place for all routes except `/` root. 

![CleanShot 2024-03-06 at 09 41 11](https://github.com/boyeonihn/manyaday/assets/95451902/209dc158-b3bd-4ec2-98ab-f694c5232f42)

Which means that in pages excluding `/`, user.value is guaranteed to exist -> but I wasn't really sure how to go about allowing `!` to be omitted here. Wasn't too sure how to even search for information, so I asked ChatGPT and this is what I got. 
![CleanShot 2024-03-06 at 09 51 04](https://github.com/boyeonihn/manyaday/assets/95451902/90341f38-97a0-4d5f-8490-bebc5dfa15df)

I feel like the second option kind of defeats the point because non-null check is still taking place. First option, I've never seen before (which isn't surprising, I'm not too familiar with TypeScript yet). Any advice?

Reply:
I see! Okay at this point, whatever Chat-GPT tells you to do, you have to do it everywhere you're using `user.value`, which is not good for your own developer experience. So what we could do is to create another composable like...

```
export const useSupabaseLoggedInUser = () => { // (or simple just useUser?)
  const user = useSupabaseUser();
  return user as Ref<SupabaseUserOrSomethingLikeThatIdoNotRememberNow>
}
```

If we cast the return value to be just like that, then it'd be easier to use it. What do you think?

Reply:
Haha, cracked up at `Ref<SupabaseUserOrSomethingLikeThatIdoNotRememberNow>` 🤣

Had to do a bit of digging to figure out what to pass for `Ref<SupabaseUserOrSth>` - looked at the github for @nuxtjs/supabase [here](https://github.com/nuxt-modules/supabase/blob/main/src/runtime/composables/useSupabaseUser.ts) I think this was what you were referring to? 
![CleanShot 2024-03-07 at 18 05 52](https://github.com/boyeonihn/manyaday/assets/95451902/13c00fdb-32cd-462e-a2df-fb526f2b23e4)

Unlike useSupabaseUser(), which seems to return either of User or null type, our composable is saying that the returned user object is always of type User from supabase-js. Is this understanding correct? 

I proceeded to create a composable `useUser` and saw it working as intended in the today page.
![CleanShot 2024-03-07 at 18 08 38](https://github.com/boyeonihn/manyaday/assets/95451902/3d4e5a32-7b58-4896-8c94-be2c67342e8f)

Reply:
yes exactly! nice digging in the supabase repo :D